### PR TITLE
Add Prometheus JMX Exporter

### PR DESCRIPTION
**Pull request by ihostage**
Inspired by https://blog.camunda.com/post/2019/06/camunda-bpm-on-kubernetes/

### CODE REVIEW

**Comment by ihostage** [comment]
Also, indentation unified for some scripts by the rule of a boy scout. 😄

File: Dockerfile [hunk 1]
```diff
@@ -42,8 +44,11 @@ ENV WAIT_FOR_TIMEOUT=30
 ENV TZ=UTC
 ENV DEBUG=false
 ENV JAVA_OPTS="-Xmx768m -XX:MaxMetaspaceSize=256m"
+ENV JMX_PROMETHEUS=false
+ENV JMX_PROMETHEUS_CONF=/camunda/javaagent/prometheus-jmx.yaml
```
Comment (cmur2) [inline Dockerfile]: It is a bit nitpicky but I'd like to canonicalise the `.yaml` extension to `.yml` as we try to do in all our systems.

File: README.md [hunk 2]
```diff
@@ -199,6 +199,14 @@ To enable JPDA inside the container you can set the environment variable
 `DEBUG=true` on startup of the container. This will allow you to connect to the
 container on port `8000` to debug your application.
 
+## Prometheus JMX Exporter
+
+To enable Prometheus JMX Exporter inside the container you can set the environment 
+variable `JMX_PROMETHEUS=true` on startup of the container. 
+This will allow you to get metrics in Prometheus format at `<host>:9404/metrics`. 
+For configuring exporter you need attach your configuration as a container volume 
+at `/camunda/javaagent/prometheus-jmx.yaml`.
```
Comment (cmur2) [inline README.md]: Extension again

**Review by cmur2 — CHANGES_REQUESTED** [review]
Thank you for your tremendously well made and valuable contribution! I like the overall idea and I'm sure it is the right tradeoff to include the JMX Prometheus Agent in terms of space and number of dependencies vs out-of-the-box usability :+1:

I also appreciate that you contribute suitable and mindful tests for this feature, definitively eases maintenance.

In the comments I would like to see one nitpicky thing changed and I'm currently also testing this on CI against EE edition to make sure it's green.

File: Dockerfile (hunk 1 above)
Comment (ihostage) [reply Dockerfile]: Done 😉

File: README.md (hunk 2 above)
Comment (ihostage) [reply README.md]: Done 😉

**Review by cmur2 — APPROVED** [review]

**Comment by cmur2** [comment]
I will merge as soon as https://travis-ci.com/camunda/docker-camunda-bpm-platform/builds/129427442 goes green.

**Comment by ihostage** [comment]
Thank a lot, @cmur2 :+1: 
May be can you review my other PR? 😄

**Comment by cmur2** [comment]
It's internally assigned to another team member who will have a look at it but it wont be forgotten :+1:

**Pull request merged**